pyanaconda/ui/gui/spokes/welcome.py

```diff
@@ -260,6 +260,7 @@ def on_locale_selection_changed(self, selection):
             lang = store[selected[0]][1]
             lang = localization.setup_locale(lang)
             self._set_lang(lang)
+            self._try_set_timezone(lang)
```

Comment:
I recommend to do this in the `apply` function after the user confirms the language selection. This could generate a huge amount of log message.